### PR TITLE
Google one tap implementation - phase 2

### PR DESCRIPTION
**Pull request by indeewari**
### Related Issues
- Issue `#1` or (None)

### Related PRs
- Related PR `#1` or (None)

### Checklist
- [ ] e2e cypress tests locally verified.
- [ ] Manual test round performed and verified.
- [ ] UX/UI review done on the final implementation.
- [ ] Documentation provided. (Add links if there are any)
- [ ] Unit tests provided. (Add links if there are any)
- [ ] Integration tests provided. (Add links if there are any)

### Security checks
- [ ] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: apps/authentication-portal/src/main/webapp/login.jsp [hunk 1]
```diff
@@ -350,22 +356,82 @@
...
-                                                title="<%=AuthenticationEndpointUtil.i18n(resourceBundle, "sign.in.with")%> <%=Encode.forHtmlAttribute(idpName)%>"
-                                            >
-                                                <img class="ui image" src="<%=Encode.forHtmlAttribute(imageURL)%>">
-                                                <span><%=AuthenticationEndpointUtil.i18n(resourceBundle, "sign.in.with")%> <%=Encode.forHtmlContent(idpName)%></span>
-                                            </button>
+                                    <% if(isGoogleIdp) {
+                                        String GOOGLE_CLIENT_ID = "";
+                                        String GOOGLE_CALLBACK_URL = "";
+                                        boolean GOOGLE_ONE_TAP_ENABLED = false;
+                                        ExternalIdPConfig externalIdPConfig = ConfigurationFacade.getInstance().getIdPConfigByName(idpName, "carbon.super");
```
Comment (indeewari) [inline apps/authentication-portal/src/main/webapp/login.jsp:363]: Have to change carbon.super to take from variable

File: apps/authentication-portal/src/main/webapp/login.jsp [hunk 2]
```diff
@@ -350,22 +356,82 @@
...
+
+                                        if(externalIdPConfig != null) {
+                                            Map<String, String> googleAuthenticatorProperties = FrameworkUtils.getAuthenticatorPropertyMapFromIdP(externalIdPConfig, idpEntry.getValue());
```
Comment (indeewari) [inline apps/authentication-portal/src/main/webapp/login.jsp:366]: Changing this to take from session utility

File: apps/authentication-portal/src/main/webapp/login.jsp [hunk 3]
```diff
@@ -350,22 +356,82 @@
...
-                                                onclick="handleNoDomain(this,
-                                                    '<%=Encode.forJavaScriptAttribute(Encode.forUriComponent(idpName))%>',
-                                                    '<%=Encode.forJavaScriptAttribute(Encode.forUriComponent(idpEntry.getValue()))%>')"
-                                                id="icon-<%=iconId%>"
-                                                title="<%=AuthenticationEndpointUtil.i18n(resourceBundle, "sign.in.with")%> <%=Encode.forHtmlAttribute(idpName)%>"
-                                            >
-                                                <img class="ui image" src="<%=Encode.forHtmlAttribute(imageURL)%>">
-                                                <span><%=AuthenticationEndpointUtil.i18n(resourceBundle, "sign.in.with")%> <%=Encode.forHtmlContent(idpName)%></span>
-                                            </button>
+                                    <% if(isGoogleIdp) {
```
Comment (thanujalk) [inline apps/authentication-portal/src/main/webapp/login.jsp:359]: Newly added code are not formatted properly.

File: apps/authentication-portal/src/main/webapp/login.jsp (hunk 1 above)
Comment (thanujalk) [inline apps/authentication-portal/src/main/webapp/login.jsp:363]: Aren't supporting to configure this for any tenant?

File: apps/authentication-portal/src/main/webapp/login.jsp [hunk 4]
```diff
@@ -499,6 +565,21 @@
...
     %>
     <script>
+        function onMoment(notification) {
+            displayGoogleSignIn(notification.isNotDisplayed() || notification.isSkippedMoment());
+        }
+
+        function displayGoogleSignIn(display) {
+            var element = document.getElementById("googleSignIn");
+            if(element != null) {
+                if(display) {
```
Comment (thanujalk) [inline apps/authentication-portal/src/main/webapp/login.jsp:575]: Format JS codes

File: apps/console/src/features/identity-providers/components/forms/authenticators/google-authenticator-form.tsx [hunk 5]
```diff
@@ -183,24 +194,62 @@ export const GoogleAuthenticatorForm: FunctionComponent<GoogleAuthenticatorFormP
             const meta: CommonAuthenticatorFormFieldMetaInterface = metadata?.properties
                 .find((meta) => meta.key === value.key);
 
+            /**
+            * Parsing string  to boolean only for Google One Tap value
+            */
+            let localValue : any;
+
+            if(value.key === IdentityProviderManagementConstants.GOOGLE_ONE_TAP_ENABLED) {
```
Comment (thanujalk) [inline apps/console/src/features/identity-providers/components/forms/authenticators/google-authenticator-form.tsx:202]: Why do we need to use identify operation here?

Comment (thanujalk) [reply apps/console/src/features/identity-providers/components/forms/authenticators/google-authenticator-form.tsx:202]: Format the code

**Review by JayaShakthi97 — APPROVED** [review]

File: apps/console/src/features/identity-providers/components/forms/authenticators/google-authenticator-form.tsx [hunk 6]
```diff
@@ -183,24 +194,62 @@ export const GoogleAuthenticatorForm: FunctionComponent<GoogleAuthenticatorFormP
...
         });
 
         setFormFields(resolvedFormFields);
         setInitialValues(resolvedInitialValues);
     }, [ originalInitialValues ]);
 
+    /**
+     * Importing all UI configurations.
+     */
+    const UIConfig: UIConfigInterface = useSelector((state: AppState) => {
```
Comment (brionmario) [inline apps/console/src/features/identity-providers/components/forms/authenticators/google-authenticator-form.tsx:233]: Let's move this to the top and group with the other `useSelector` hooks.

**Review by brionmario — APPROVED** [review]

**Comment by indeewari** [comment]
Closing due to PAT issue of the branch. PR merged on https://github.com/wso2/identity-apps/pull/3480

**Pull request closed without merging**